sqlmesh/core/config/scheduler.py

```diff
@@ -96,6 +95,9 @@ def create_plan_evaluator(self, context: Context) -> PlanEvaluator:
             notification_target_manager=context.notification_target_manager,
         )
 
+    def get_default_catalog(self, context: Context) -> t.Optional[str]:
+        return context.engine_adapter.default_catalog
```

Comment:
It occurred to me now: shall we raise on the client side as well when the variable is not set?

Reply:
I updated this for the Airflow schedulers to raise. Builtin still supports not having a default catalog.